validator/db/kv/genesis.go

```diff
@@ -0,0 +1,36 @@
+package kv
+
+import (
+	"context"
+	"fmt"
+
+	bolt "go.etcd.io/bbolt"
+)
+
+// SaveGenesisValidatorsRoot saves the genesis validator root to db.
+func (s *Store) SaveGenesisValidatorsRoot(ctx context.Context, genValRoot []byte) error {
+	err := s.db.Update(func(tx *bolt.Tx) error {
+		bkt := tx.Bucket(genesisInfoBucket)
+		enc := bkt.Get(genesisValidatorsRootKey)
+		if len(enc) != 0 {
+			return fmt.Errorf("cannot overwite existing genesis validators root: %#x", enc)
```

Comment:
i think it should be fatal

Reply:
The fatal log should be by the caller of this function